dataland-frontend/src/components/forms/parts/fields/YesNoFormField.vue

```diff
@@ -1,33 +1,131 @@
+        :more-than-one-document-allowed="false"
+      />
+      <FormKit v-if="yesSelected" type="group" name="dataSource">
+        <FormKit type="hidden" name="name" v-model="documentName" />
+        <FormKit
+          type="text"
+          name="reference"
+          v-model="documentReference"
+          validation="required"
+          validation-label="If 'Yes' is selected an uploaded document"
+          :outer-class="{ 'hidden-input': true }"
+        />
+      </FormKit>
+    </FormKit>
+
+    <RadioButtonsFormElement
+      v-else
+      :name="name"
+      :validation="validation"
+      :validation-label="validationLabel ?? displayName"
+      :options="[
+        {
+          label: 'Yes',
+          value: 'Yes',
+        },
+        {
+          label: 'No',
+          value: 'No',
+        },
+      ]"
+    />
+  </div>
 </template>
 
 <script lang="ts">
 import { defineComponent } from "vue";
-import RadioButtonsFormField from "@/components/forms/parts/fields/RadioButtonsFormField.vue";
-import { FormFieldProps } from "@/components/forms/parts/fields/FormFieldProps";
+import { YesNoFormFieldProps } from "@/components/forms/parts/fields/FormFieldProps";
+import RadioButtonsFormElement from "@/components/forms/parts/elements/basic/RadioButtonsFormElement.vue";
+import UploadFormHeader from "@/components/forms/parts/elements/basic/UploadFormHeader.vue";
+import UploadDocumentsForm from "@/components/forms/parts/elements/basic/UploadDocumentsForm.vue";
+import { DocumentToUpload } from "@/utils/FileUploadUtils";
 
 export default defineComponent({
   name: "YesNoFormField",
-  components: { RadioButtonsFormField },
+  components: { RadioButtonsFormElement, UploadFormHeader, UploadDocumentsForm },
   inheritAttrs: false,
-  props: FormFieldProps,
+  props: YesNoFormFieldProps,
+  data() {
+    return {
+      yesSelected: false,
+      referencedDocument: {} as DocumentToUpload,
+      documentName: "",
+      documentReference: "",
+    };
+  },
+
+  emits: ["documentUpdated"],
+  watch: {
+    yesSelected() {
+      this.deleteDocument();
+    },
+    documentName() {
+      this.updateFileUploadFiles();
+    },
+  },
+  methods: {
+    /**
+     * Sets the value yesSelected to true when "Yes" is selected
+     * @param event the "Yes" / "No" selection event
+     */
+    setDocumentRequired(event: Event) {
+      this.yesSelected = (event as unknown as string) === "Yes";
+    },
+
+    /**
+     * If "No" is reselected after one has uploaded a file, this handles removing the file(s) and clearing
+     * the certificate list.
+     */
+    deleteDocument() {
+      if (!this.yesSelected) {
+        (this.$refs.uploadDocumentsForm.removeAllDocuments as () => void)();
+      }
+    },
+
+    /**
+     * Emits event that selected document changed
+     * @param updatedDocuments the updated documents that are currently selected (only one in this case)
+     */
+    handleDocumentUpdatedEvent(updatedDocuments: DocumentToUpload[]) {
```

Comment:
I'm a bit confused between Reports, Files and Documents. Are those three separate things or different names for the same?

Reply:
They extend each other. The hierarchy is File -> Document -> Report.
So every document is a file and every report is a document. But not every document is a report.

Reply:
Same as the other commentar. We agreed to have a short team meeting to discuss the file hierachy.

Reply:
We discussed that we want this structure also for other occurences of reports. We create a new story for this.